### PR TITLE
(#41) read keystore config from application.yml, add keystore file to resources

### PR DESCRIPTION
**Pull request by rishabGhoshtw**

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 14
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/consent/ConsentConfiguration.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5416385&bid=17758690)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 7
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/consent/ConsentConfiguration.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5416385&bid=17758690)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/consent/ConsentConfiguration.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5416385&bid=17758690)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/consent/ConsentConfiguration.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5416385&bid=17758690)

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -53,3 +53,10 @@ spring:
   rabbitmq:
     host: ${RABBITMQ_HOST}
     port: ${RABBITMQ_PORT}
+
+keyStore:
+  filePath: "./src/main/resources/consent-manager-keystore.p12"
```
Comment (ganesanarun) [inline src/main/resources/application.yml:58]: Configurable. and comments about how to generate one, and one in test/root which can be used for local development.

File: src/main/java/in/projecteka/consentmanager/consent/ConsentConfiguration.java [hunk 2]
```diff
@@ -8,26 +8,42 @@
...
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.web.reactive.function.client.WebClient;
 
-import java.security.KeyPair;
-import java.security.KeyPairGenerator;
-import java.security.PublicKey;
+import java.io.File;
+import java.io.FileInputStream;
+import java.security.*;
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/consent/ConsentConfiguration.java:28]: wildcard imports/

File: src/main/java/in/projecteka/consentmanager/consent/ConsentConfiguration.java [hunk 3]
```diff
@@ -140,13 +156,22 @@ public ConsentRequestNotificationListener consentRequestNotificationListener(
...
     @Bean
     public KeyPair keyPair() {
-        KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
-        keyGen.initialize(2048);
-        return keyGen.generateKeyPair();
+        KeyStore keyStore = KeyStore.getInstance(keyStoreType);
+        File file = new File(keyStoreFilePath);
+        char[] pwdArray = keyStorePassword.toCharArray();
+        keyStore.load(new FileInputStream(file), pwdArray);
+        Certificate certificate = keyStore.getCertificate(keyStoreAlias);
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/consent/ConsentConfiguration.java:163]: same key used for pin verification and consent artefact sigining?

File: src/main/resources/application.yml (hunk 1 above)
Comment (rishabGhoshtw) [reply src/main/resources/application.yml:58]: why should we use test/root keystore for local development??

File: src/main/java/in/projecteka/consentmanager/consent/ConsentConfiguration.java (hunk 3 above)
Comment (rishabGhoshtw) [reply src/main/java/in/projecteka/consentmanager/consent/ConsentConfiguration.java:163]: No. pin verification key is still getting generated dynamically. I will quickly replace it

File: src/main/resources/application.yml (hunk 1 above)
Comment (ganesanarun) [reply src/main/resources/application.yml:58]: if I don't want to create one by myself :)

Comment (ganesanarun) [reply src/main/resources/application.yml:58]: That's okay if you don't add it now.

You need to figure out how we can add the keystore into dockerimage though

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/test/java/in/projecteka/consentmanager/common/KeyPairConfigTest.java  1
- src/main/java/in/projecteka/consentmanager/common/KeyPairConfig.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5416385&bid=17758690)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/common/KeyPairConfig.java  3
- src/test/java/in/projecteka/consentmanager/common/KeyPairConfigTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5416385&bid=17758690)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/common/KeyPairConfig.java  3
- src/test/java/in/projecteka/consentmanager/common/KeyPairConfigTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5416385&bid=17758690)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Complexity increasing per file
==============================
- src/test/java/in/projecteka/consentmanager/common/KeyPairConfigTest.java  1
- src/main/java/in/projecteka/consentmanager/common/KeyPairConfig.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5416385&bid=17758690)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Complexity increasing per file
==============================
- src/test/java/in/projecteka/consentmanager/common/KeyPairConfigTest.java  1
- src/main/java/in/projecteka/consentmanager/common/KeyPairConfig.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5416385&bid=17758690)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Complexity increasing per file
==============================
- src/test/java/in/projecteka/consentmanager/common/KeyPairConfigTest.java  1
- src/main/java/in/projecteka/consentmanager/common/KeyPairConfig.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5416385&bid=17758690)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Complexity increasing per file
==============================
- src/test/java/in/projecteka/consentmanager/common/KeyPairConfigTest.java  1
- src/main/java/in/projecteka/consentmanager/common/KeyPairConfig.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5416385&bid=17758690)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/common/KeyPairConfig.java  3
- src/test/java/in/projecteka/consentmanager/common/KeyPairConfigTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5416385&bid=17758690)

**Comment by ganesanarun** [comment]
Need to setup the NCG environment there.

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- src/test/java/in/projecteka/consentmanager/common/KeyPairConfigTest.java  1
- src/main/java/in/projecteka/consentmanager/common/KeyPairConfig.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5416385&bid=17758690)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/common/KeyPairConfig.java  3
- src/test/java/in/projecteka/consentmanager/common/KeyPairConfigTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5416385&bid=17758690)

**Pull request merged**